airflow/providers/google/cloud/utils/credentials_provider.py

```diff
@@ -241,7 +241,14 @@ def get_credentials_and_project_id(
         project_id = credentials.project_id
 
     if delegate_to:
-        credentials = credentials.with_subject(delegate_to)
+        if hasattr(credentials, 'with_subject'):
```

Comment:
Can you add tests?

Reply:
Ok. Should the test case be if `delegate_to` is passed, the Exception is raised?

Reply:
Yes. For this case.

Reply:
Hmm, I'm struggling to work out how to make sure `google.auth.default()` returns a credentials object without the `with_subject` attribute.

Reply:
I will look at it and prepare an example for you. I suspect that mock + spec/autospec will allow this.

Reply:
```
from google.auth import compute_engine
```
```
    @mock.patch('google.auth.default')
    def test_get_credentials_and_project_id_with_default_auth_and_unsupported_delegate(self, mock_auth_default):
        mock_credentials = mock.MagicMock(spec=compute_engine.Credentials)
        mock_auth_default.return_value = (mock_credentials, self.test_project_id)

        with self.assertRaisesRegex(AirflowException, re.escape(
            "The `delegate_to` parameter cannot be used here as the current authentication method does not "
            "support account impersonate. Please use service-account for authorization."
        )):
            get_credentials_and_project_id(delegate_to="USER")
```
Can you check it?